### PR TITLE
[core] In ACLiC, do not reload loaded libraries.

### PR DESCRIPTION
**Pull request by vgvassilev**
TCling::Load reloads the library the `system` optional parameter is false. If the library is not marked as `system` TCling reloads it if was alread loaded. That is, it dlcloses and dlopens the library instead of doing nothing.

There is no point in reloading our library dependencies.

The regression was introduced in 9b6df8c originated from https://github.com/root-project/root/commit/e649f59e3160f829bc1a4813f8aa73bdd16b6cff

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
The failures are present in the last nightly builds.

File: core/base/src/TSystem.cxx [hunk 1]
```diff
@@ -3379,7 +3379,7 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
       // the ACLiC library.
       if (useCxxModules && !produceRootmap) {
          std::function<bool(const char *)> LoadLibF = [](const char *dep) {
-            return gInterpreter->Load(dep, /*system*/ false) >= 0;
+            return gSystem->Load(dep) >= 0;
```
Comment (pcanal) [inline core/base/src/TSystem.cxx:3382]: This is likely a step in the right direction but more needs to be done as ```gSystem->Load``` is implemented in term of ```gInterpreter->Load```.  So it is really the behavior in gInterpreter->Load that have been (very backwardly incompatible) altered (it nows reload shared libraries and re-execute their static initialization),

Comment (vgvassilev) [reply core/base/src/TSystem.cxx:3382]: Looks like whenever we want to permanently load (and not reload if already loaded) the library we need to set the `system` optional parameter to `true`: https://github.com/root-project/root/blob/master/core/metacling/src/TCling.cxx#L3355

File: core/metacling/src/TCling.cxx [hunk 2]
```diff
@@ -1832,9 +1832,7 @@ void TCling::LoadPCM(std::string pcmFileNameFullPath)
       TMemFile pcmMemFile(RDictFileOpts.c_str(), range);
 
       LoadPCMImpl(pcmMemFile);
-      // FIXME: Uncomment this when we implement UnLoadPCM as per
-      // root-project/root#5420
-      //fPendingRdicts.erase(pendingRdict);
+      fPendingRdicts.erase(pendingRdict);
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx:1835]: This is a bit unrelated to the PR and the comment is still valid.  So I would skip that.

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx:1835]: It is, if I remove the comments without the changes to TSystem, tests will start to fail. With the change in TSystem uncommenting these lines should work just fine. That code actually detected the issue.

Comment (pcanal) [reply core/metacling/src/TCling.cxx:1835]: Yes.  That is what I expect:

Without this PR, reloading libraries is exercised (a lot).

That the code in TCling.cxx has no negative effect only means that after this PR, the test are no longer reloading libraries that have a PCM (or something like) and thus unloading library and that line are no longer exercise.

Unless I am mistaken it still means that we need to implement UnloadPCM at some point, don't we?

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx:1835]: Yes, the urgency however changes a bit.

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
Failures are present in the nightlies so they are independent on this PR.
